drivers/ethernet/eth_adin2111.c

```diff
@@ -938,7 +938,7 @@ static const struct ethernet_api adin2111_port_api = {
 #define ADIN2111_XSTR(x)	ADIN2111_STR(x)
 
 #define ADIN2111_MDIO_PHY_BY_ADDR(adin_n, phy_addr)						\
-	DEVICE_DT_GET(DT_CHILD(DT_INST_CHILD(adin_n, mdio), phy_##phy_addr))
+	DEVICE_DT_GET(DT_CHILD(DT_INST_CHILD(adin_n, mdio), ethernet_phy_##phy_addr))
```

Comment:
That would require modification of an overlay here #63001

Reply:
Updated my comments in the migration guide to mention this